src/components/Inputs/TextInput/TextInput.tsx

```diff
@@ -392,88 +394,120 @@ export const TextInput: FC<TextInputProps> = React.forwardRef(
+                                        <Icon
+                                            {...iconProps}
+                                            path={iconProps.path}
+                                            size={inputSizeToIconSizeMap.get(
+                                                mergedSize
+                                            )}
+                                        />
                                     )}
-                                />
+                                    {iconProps.imageSrc && !iconProps.path && (
+                                        <img
+                                            aria-hidden={iconProps.ariaHidden}
+                                            alt={iconProps.alt}
+                                            id={iconProps.id}
+                                            src={iconProps.imageSrc}
+                                        />
+                                    )}
+                                </div>
                             )}
-                            {iconProps.imageSrc && !iconProps.path && (
-                                <img
-                                    aria-hidden={iconProps.ariaHidden}
-                                    alt={iconProps.alt}
-                                    id={iconProps.id}
-                                    src={iconProps.imageSrc}
+                            {iconButtonProps && (
+                                <DefaultButton
+                                    allowDisabledFocus={
+                                        iconButtonProps.allowDisabledFocus
+                                    }
+                                    ariaLabel={iconButtonProps.ariaLabel}
+                                    checked={iconButtonProps.checked}
+                                    classNames={iconButtonClassNames}
+                                    disabled={
+                                        iconButtonProps.disabled ||
+                                        mergedDisabled
+                                    }
+                                    iconProps={{
+                                        path: iconButtonProps.iconProps.path,
+                                    }}
+                                    id={iconButtonProps.id}
+                                    onClick={iconButtonProps.onClick}
+                                    size={inputSizeToButtonSizeMap.get(
+                                        mergedSize
+                                    )}
+                                    htmlType={iconButtonProps.htmlType}
                                 />
                             )}
+                            {clearable &&
+                                clearButtonShown &&
+                                !numbersOnly &&
+                                htmlType !== 'number' && (
+                                    <DefaultButton
```

Comment:
was this clearable button always outside of the input? It seems a bit oddly placed

Reply:
yes, i did not change the order.